nimare/meta/utils.py

```diff
@@ -130,19 +130,22 @@ def _convolve_sphere(kernel, peaks):
         )
 
         all_spheres = all_spheres[idx, :]
+
         if sum_overlap:
             counts = counts[idx]
 
-        ma_values = np.zeros(shape)
-        ma_values[tuple(all_spheres.T)] = counts
-        # Set voxel outside the mask to zero.
-        ma_values[~mask_data] = 0
+        sphere_idx_inside_mask = np.where(mask_data[tuple(all_spheres.T)])[0]
+        sphere_idx_filtered = all_spheres[sphere_idx_inside_mask, :].T
+        nonzero_idx = tuple(sphere_idx_filtered)
 
-        nonzero_idx = np.where(ma_values > 0)
+        if sum_overlap:
+            nonzero_to_append = counts[sphere_idx_inside_mask]
+        else:
+            nonzero_to_append = np.ones((len(sphere_idx_inside_mask),)) * counts
```

Comment:
```suggestion
            nonzero_to_append = np.ones((len(sphere_idx_inside_mask),)) * value
```
With these changes, I think here we could use `value` and remove `counts = value` from line 125 so that counts will always be a list.